### PR TITLE
Deprecate configtest and add validate command

### LINKED ISSUE #2421: Consul configtest Improvements
The `consul configtest` command could use a few small improvements:
- [ ] Rename to `configtest` to `validate`. In an effort to provide consistency across our products, we should rename configtest to validate. This matches parity with Terraform and Nomad, and it also gives us a slightly more descriptive name. We should [hide the configtest command](https://github.com/mitchellh/cli/blob/074e243518e473e0887217008dd013e07755be8c/help.go#L63-L79) in 0.8, then remove it in 0.9.
- [ ] Accept either a file path or directory, but Consul should determine that. The current behavior is to specify `-config-file` or `-config-dir`. This makes for a slightly less exciting UX than other tools, which can accept either a directory or a single file as an argument. I would recommend the UX:
  
  ```
  $ consul validate [OPTIONS] FILE_OR_DIRECTORY
  ```
  
  Consul can intelligently inspect the file statistics and see if it's a directory or a file. Keep `-config-file` and `-config-dir` flags around as hidden, remove in 0.9.
- [ ] Return output if the validation is successful. The current behavior is to just exit with a 0 exit code. This is great for scripting, but makes humans confused. The CLI should output "configuration is valid" or some kind of success message on success. This can be coupled with a new -quiet flag which suppresses said message.

### PR DESCRIPTION
**Pull request by kyhavlov**
Hides the configtest command, adds a deprecation notice, and adds the validate command with the old configtest flags as hidden flags.

Closes #2421

### CODE REVIEW

**Comment by slackpad** [comment]
LGTM with one addition - missed the success message and `-quiet` part of #2421.

**Comment by slackpad** [comment]
LGTM

File: command/validate.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+	return strings.TrimSpace(helpText)
+}
+
+func (c *ValidateCommand) Run(args []string) int {
+	var configFiles []string
+	var quiet bool
+
+	f := c.Command.NewFlagSet(c)
+	f.Var((*agent.AppendSliceValue)(&configFiles), "config-file",
```
Comment (sethvargo) [inline command/validate.go]: @kyhavlov I still think it'd be a better UX to just accept one option here and determine if it's a file or folder appropriately. I don't see a reason why we need to treat them separately.

Comment (kyhavlov) [inline command/validate.go]: I left those in as hidden flags to stay compatible with configtest; everything (args, `-config-file`, `config-dir`) gets added to `configFiles` which works with files or directories. Do you think we should just remove the flags from validate altogether?

Comment (sethvargo) [inline command/validate.go]: Ah sorry. I didn't see the "hidden" call below. +1 to BC! Great work on this and sorry for the confusion

**Comment by mterron** [comment]
consul validate seem to fail is there's no 'data_dir' in the config file even though you could just use a cli option to provide that parameter.

**Pull request merged**